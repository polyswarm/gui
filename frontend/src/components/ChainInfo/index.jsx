// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty Manager imports
// Component imports

class ChainInfo extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
        <div />
    )
  }

}
ChainInfo.proptypes = {

}
export default ChainInfo;