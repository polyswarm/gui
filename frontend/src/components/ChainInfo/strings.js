const strings = {
    current: 'Current Balance',
    adjusted: 'Balance after Transfer'
};
export default strings;