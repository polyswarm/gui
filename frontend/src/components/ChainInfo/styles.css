.ChainInfo {
    background-color: rgb(46, 255, 109);
    border-radius: 4px;
    box-shadow: 0 1px 2px #000000;
    box-sizing: border-box;
    color: #8500ff;
    display: flex;
    flex-direction: column;
    margin: 15px 0 15px 0;
    padding: 5px 20px 5px 20px;
    width: 45%;
}

.ChainInfo-Title > h2 {
    margin: 0 0 8px 0;
}

.ChainInfo-Content > ul > li {
    margin: 0 0 8px 0;
}

.ChainInfo-Content > ul {
    list-style: none;
    padding-left: 0;
}