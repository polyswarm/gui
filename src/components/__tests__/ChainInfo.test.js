import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import ChainInfo from '../ChainInfo';

it('renders without crashing', () => {
  const wrapper = render(<ChainInfo balance={0} transfer={0}/>);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('displays the name of the networks', () => {
  const wrapper = render(<ChainInfo homeName={'home'} sideName={'side'}/>);
  
  expect(wrapper.find('li').first().text()).toEqual('home: ');
  expect(wrapper.find('li').last().text()).toEqual('side: ');
});

it('displays the current balances', () => {
  const wrapper = render(<ChainInfo homeName={'home'}
    homeBalance={1}
    sideName={'side'}
    sideBalance={2}/>);
  
    expect(wrapper.find('li').first().text()).toEqual('home: 1');
    expect(wrapper.find('li').last().text()).toEqual('side: 2');
});

it('displays the title', () => {
  const wrapper = render(<ChainInfo title={'Title'}/>);

  expect(wrapper.find('h2').text()).toEqual('Title');
});