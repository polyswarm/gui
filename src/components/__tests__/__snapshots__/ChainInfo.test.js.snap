// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="ChainInfo"
>
  <div
    class="ChainInfo-Title"
  >
    <h3>
      
    </h3>
  </div>
  <div
    class="ChainInfo-Content"
  >
    <p>
      
    </p>
    <ul>
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          
        </p>
        <p
          class="StatContent"
        >
          undefined NCT
        </p>
      </li>
      <li
        class="StatRow-Vertical"
      >
        <p
          class="StatTitle"
        >
          
        </p>
        <p
          class="StatContent"
        >
          undefined NCT
        </p>
      </li>
    </ul>
  </div>
</div>
`;
